libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -134,15 +135,26 @@ type segmentHeader struct {
 	// If the segment file has not been completely written, this field is zero.
 	// Only present in schema version >= 1.
 	frameCount uint32
+
+	// options holds flags to enable features, for example encryption.
+	options uint8
```

Comment:
Any reason not to just make this larger for more future proofing? Probably nobody will notice 1-3 extra bytes in each segment file depending on uint16 or uint32 as the larger size.

Reply:
I waffled on this.  In the end I picked uint8 because I couldn't think of even 8 options, but we could easily go with uint32, that would match version & count.